### PR TITLE
Ory Kratos updated to v1.2; Session Refresh fixed by proxying fresh 🍪

### PR DESCRIPTION
**Pull request by me-andre**
…ed cookies

### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/core/authentication/authentication.service.ts [hunk 1]
```diff
@@ -176,48 +177,78 @@ export class AuthenticationService {
...
+      return true; // Set to -1 if specified as lifespan in config, meaning it can be extended at any time
+    }
 
     const expiry = new Date(session.expires_at);
-    return Date.now() >= expiry.getTime() - this.extendAfter;
+    return Date.now() >= expiry.getTime() - this.extendSessionMinRemainingTTL;
+  }
+
+  private parseEarliestPossibleExtend(
+    earliestPossibleExtend: unknown
```
Comment (hero101) [inline src/core/authentication/authentication.service.ts:236]: can you narrow down the type?

File: src/core/middleware/session.extend.middleware.ts [hunk 2]
```diff
@@ -98,24 +87,12 @@ export class SessionExtendMiddleware implements NestMiddleware {
...
-        return next();
-      }
-
-      res.cookie(this.SESSION_COOKIE_NAME, orySessionId, {
-        path: '/',
-        sameSite: 'lax',
-        expires: new Date(newExpiry),
-        httpOnly: true,
-        encode: val => val, // skip encoding; pass the value as is
+      const { headers } = await this.kratosClient.toSession({
```
Comment (me-andre) [inline src/core/middleware/session.extend.middleware.ts:98]: ⚠️Add a comment why proxying headers is needed and we can't just extend the old cookie.

File: src/core/middleware/session.extend.middleware.ts [hunk 3]
```diff
@@ -58,9 +58,7 @@ export class SessionExtendMiddleware implements NestMiddleware {
 
     let session: Session;
     try {
-      session = await getSession(this.kratosClient, {
-        authorization,
-      });
+      session = getSessionFromJwt(authorization);
```
Comment (hero101) [inline src/core/middleware/session.extend.middleware.ts:61]: this function requires the token, and you are passing the whole value of the header.
`getSession` was able to parse the header, but this one doesn't

**Review by hero101 — CHANGES_REQUESTED** [review]
Will need to test locally.

File: src/core/authentication/authentication.service.ts (hunk 1 above)
Comment (me-andre) [reply src/core/authentication/authentication.service.ts:236]: It is narrowed down already (config gives `any`). From here on we operate on the narrowest possible type (hence the `typeof` below).

File: src/core/middleware/session.extend.middleware.ts (hunk 3 above)
Comment (hero101) [reply src/core/middleware/session.extend.middleware.ts:61]: Turns out that it actually can parse the whole header.

File: src/core/middleware/session.extend.middleware.ts [hunk 4]
```diff
@@ -70,22 +68,13 @@ export class SessionExtendMiddleware implements NestMiddleware {
...
-        });
-
-      if (!newSession) {
-        this.logger.warn?.(
-          'New session cookie not sent: new session not found',
-          LogContext.AUTH_TOKEN
+      try {
+        await this.authService.extendSession(session);
+      } catch (error) {
+        this.logger.error(
```
Comment (hero101) [inline src/core/middleware/session.extend.middleware.ts:74]: when this fails the code continues and sends a cookie everytime

File: src/core/authentication/authentication.service.ts [hunk 5]
```diff
@@ -176,48 +177,78 @@ export class AuthenticationService {
...
+  ): number | undefined {
+    /**
+     * If you need high flexibility when extending sessions, you can set earliest_possible_extend to lifespan,
```
Comment (hero101) [inline src/core/authentication/authentication.service.ts:239]: I believe this must be a description to the variable in the config.

**Review by hero101 — APPROVED** [review]

File: src/core/authentication/authentication.service.ts (hunk 5 above)
Comment (me-andre) [reply src/core/authentication/authentication.service.ts:239]: done

**Pull request merged**